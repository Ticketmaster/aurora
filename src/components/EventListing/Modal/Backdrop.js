import styled from 'styled-components';
import Flex from "../../Flex";

const Backdrop = styled(Flex)`
  align-items: center;
  background-color: rgba(0,0,0,.5);
  height: 100%;
  justify-content: center;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 99; 
`;

export default Backdrop;
