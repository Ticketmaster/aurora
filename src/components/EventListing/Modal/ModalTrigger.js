import React from 'react';

const ModalTrigger = ({
  buttonRef,
  children,
  onOpen,
  onOpenSuccess
}) => (
  <React.Fragment>
    {children(onOpen, buttonRef, onOpenSuccess)}
  </React.Fragment>
);

export default ModalTrigger;