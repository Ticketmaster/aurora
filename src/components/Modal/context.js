import React from "react";

const { Provider, Consumer } = React.createContext();

export { Provider as ModalProvider };
export { Consumer as ModalConsumer };

export const withModal = Component => props => (
  <Consumer>{value => <Component modal={value} {...props} />}</Consumer>
);
