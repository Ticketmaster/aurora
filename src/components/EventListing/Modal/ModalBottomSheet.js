import React from 'react';
import ReactDOM from 'react-dom';
import styled from "styled-components";

import { CSSTransition, TransitionGroup } from "react-transition-group";
import { constants, spacing } from "../../../theme";

import Flex from "../../Flex";
import Backdrop from "./Backdrop";

const ModalBottomSheet = ({
  ariaLabel,
  buttonRef,
  content,
  children,
  modalRef,
  onClickAway,
  onClose,
  isOpen,
  role = 'dialog'
}) => ReactDOM.createPortal(
    <TransitionGroup component="div">
      <CSSTransition
        classNames="bottom-sheet"
        timeout={300}
        in={isOpen}
        appear
        unmountOnExit
        mountOnEnter
      >
        <Backdrop
          column
          className="c-modal-cover"
          role={role}
          aria-label={ariaLabel}
          aria-modal="true"
          onClick={onClickAway}
        > 
          <Content
            column
            className="c-modal"
            ref={modalRef}
          >
            <Flex>
              <button
                aria-labelledby="close-modal"
                className="c-modal__close"
                ref={buttonRef}
                onClick={onClose}
              >
                <span id="close-modal" className="u-hide-visually">Close Modal</span>
                <svg viewBox="0 0 40 40" className="c-modal__close-icon">
                  <path d="M 10,10 L 30,30 M 30,10 L 10,30" />
                </svg>

              </button>
            </Flex>

            <div className="c-modal__body">
              {children}
            </div>
            
          </Content>
        </Backdrop>
      </CSSTransition>
    </TransitionGroup>
    ,
    document.body
  )

export default ModalBottomSheet;


const Content = styled.div`
  background-color: white;
  width: 100%;
  z-index: 8;
  max-height: calc(100% - 60px);
  border-top-left-radius: ${constants.borderRadius.large};
  border-top-right-radius: ${constants.borderRadius.large};
  position: fixed;
  bottom: 0;
  left: 0;
  padding-right: ${spacing.cozy};
  padding-left: ${spacing.cozy};

  .bottom-sheet-enter & {
    transform: translateY(100%);
  }

  .bottom-sheet-leave &,
  .bottom-sheet-exit & {
    transform: translateY(0);
  }

  .bottom-sheet-appear & {
    transform: translateY(100%);
  }

  .bottom-sheet-enter.bottom-sheet-enter-active & {
    transform: translateY(0);
    transition: transform 300ms ease-in-out;
  }

  .bottom-sheet-leave.bottom-sheet-leave-active &,
  .bottom-sheet-exit.bottom-sheet-exit-active & {
    transform: translateY(100%);
    transition: transform 300ms ease-in-out;
  }

  .bottom-sheet-appear.bottom-sheet-appear-active & {
    transform: translateY(0);
    transition: transform 300ms ease-in-out;
  }
`;



