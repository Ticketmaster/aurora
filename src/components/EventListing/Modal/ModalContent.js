import React from 'react';
import ReactDOM from 'react-dom';
import { CSSTransition, TransitionGroup } from "react-transition-group";

import styled from "styled-components";
import Flex from "../../Flex";
import Backdrop from "./Backdrop";

const ModalContent = ({
  ariaLabel,
  buttonRef,
  content,
  children,
  modalRef,
  onClickAway,
  onClose,
  isOpen,
  role = 'dialog'
}) => ReactDOM.createPortal(
    <TransitionGroup component="div">
      <CSSTransition
        classNames="modal"
        timeout={{
          enter: 300,
          exit: 100
        }}
        in={isOpen}
        appear
        unmountOnExit
        mountOnEnter
      >
        <Backdrop
          column
          className="c-modal-cover"
          role={role}
          aria-label={ariaLabel}
          aria-modal="true"
          onClick={onClickAway}
        > 
          <Content
            column
            className="c-modal"
            ref={modalRef}
          >
            <Flex>
              <button
                aria-labelledby="close-modal"
                className="c-modal__close"
                ref={buttonRef}
                onClick={onClose}
              >
                <span id="close-modal" className="u-hide-visually">Close Modal</span>
                <svg viewBox="0 0 40 40" className="c-modal__close-icon">
                  <path d="M 10,10 L 30,30 M 30,10 L 10,30" />
                </svg>

              </button>
            </Flex>

            <div className="c-modal__body">
              {children}
            </div>
            
          </Content>
        </Backdrop>
      </CSSTransition>
    </TransitionGroup>
    ,
    document.body
  )

export default ModalContent;


const Content = styled(Flex)`
    background-color: rgba(255,255,255,1);
    border-radius: 4px;
    border: solid 1px rgba(0,0,0,0.04);
    box-shadow: 0 16px 16px 0 rgba(0,0,0,0.06), 0 0 16px 0 rgba(0,0,0,0.12);
    max-height: calc(100% - 96px);
    max-width: 640px;
    min-width: 400px;
    overflow-x: hidden;
    overflow-y: auto;
    padding: 0;
    position: relative;
    width: fit-content;
    z-index: 100;

    .modal-enter & {
      opacity: 0;
      transform: scale(0.7);
    }

    .modal-leave &,
    .modal-exit & {
      opacity: 1;
      transform: scale(1);
    }
    .modal-appear & {
      opacity: 0;
      transform: scale(0.7);
    }
    .modal-enter.modal-enter-active & {
      opacity: 1;
      transform: scale(1);
      transition: all 0.3s ease-in-out;
    }
    .modal-leave.modal-leave-active &,
    .modal-exit.modal-exit-active & {
      opacity: 0;
      transition: all 0.1s ease-in-out;
      transform: scale(0.7);
    }
    .modal-appear.modal-appear-active & {
      opacity: 1;
      transform: scale(1);
      transition: all 0.3s ease-in-out;
    }

`;


// accessability
{/* <ReactFocusTrap
      tag="aside"
      focusTrapOptions={{ onDeactivate: onClose }}
      className="c-modal-cover"
      role={role}
      aria-label={ariaLabel}
      aria-modal="true"
      onClick={onClickAway}
    >
      <div className="c-modal" ref={modalRef}>
        <button className="c-modal__close" aria-labelledby="close-modal" onClick={onClose} ref={buttonRef}>
          <span id="close-modal" className="u-hide-visually">Close Modal</span>
          <svg viewBox="0 0 40 40" className="c-modal__close-icon">
            <path d="M 10,10 L 30,30 M 30,10 L 10,30"></path>
          </svg>
        </button>
        <div className="c-modal__body">{content}</div>
      </div>
    </ReactFocusTrap>,
    document.body
  ) */}