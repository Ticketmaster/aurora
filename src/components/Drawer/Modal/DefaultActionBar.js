import React from "react";
import PropTypes from "prop-types";

import { colors } from "../../theme/index";

import CrossIcon from "../Icons/Cross";

import { withModal } from "./context";

import { CloseButtonContainer } from "./DefaultActionBar.styles";

// TODO: allow to set up the title and background
export const ModalActionBar = ({ modal: { closeModal }, iconButtonProps }) => (
  <CloseButtonContainer>
    <CrossIcon
      size={12}
      style={{ pointerEvent: "none" }}
      color={colors.onyx.base}
      className="button--close"
      aria-label="Close Modal"
      role="button"
      onClick={closeModal}
      {...iconButtonProps}
    />
  </CloseButtonContainer>
);

ModalActionBar.propTypes = {
  modal: PropTypes.shape({
    closeModal: PropTypes.func.isRequired
  }).isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  iconButtonProps: PropTypes.object
};

ModalActionBar.defaultProps = {
  iconButtonProps: {}
};

export default withModal(ModalActionBar);
