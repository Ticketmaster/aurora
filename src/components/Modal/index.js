import React from "react";
import PropTypes from "prop-types";
import { CSSTransition } from "react-transition-group";

import {
  ModalContainer,
  ModalContent,
  ActionBar,
  BottomActionBar
} from "./index.styles";

import Backdrop from "../Backdrop";

import { ModalProvider } from "./context";
import {
  MODAL_SIZE_SMALL,
  MODAL_SIZE_MEDIUM,
  MODAL_SIZE_LARGE,
  MODAL_SIZE_XLARGE,
  getModalSize
} from "./size";
import { withDeviceSize } from "../DeviceSize/Context";
import {
  getContentHeight,
  getBottomActionBarShadow,
  getActionBarShadow,
  isRequestCloseApproved
} from "./helper";

class Modal extends React.Component {
  static propTypes = {
    actionBar: PropTypes.node,
    bottomActionBar: PropTypes.node,
    children: PropTypes.node.isRequired,
    deviceSize: PropTypes.shape({
      isXSmall: PropTypes.bool,
      isSmall: PropTypes.bool,
      isMedium: PropTypes.bool,
      isLarge: PropTypes.bool,
      isXLarge: PropTypes.bool
    }).isRequired,
    gutters: PropTypes.bool,
    // This is used inside the `getDerivedStateFromProps`
    // eslint-disable-next-line react/no-unused-prop-types
    isOpened: PropTypes.bool,
    size: PropTypes.oneOf([
      MODAL_SIZE_SMALL,
      MODAL_SIZE_MEDIUM,
      MODAL_SIZE_LARGE,
      MODAL_SIZE_XLARGE
    ]),
    onRequestClose: PropTypes.func,
    onScroll: PropTypes.func,
    /* eslint-disable react/forbid-prop-types */
    containerProps: PropTypes.object,
    actionBarProps: PropTypes.object,
    contentProps: PropTypes.object,
    bottomActionBarProps: PropTypes.object
    /* eslint-enable react/forbid-prop-types */
  };

  static defaultProps = {
    actionBar: null,
    bottomActionBar: null,
    gutters: true,
    isOpened: true,
    size: MODAL_SIZE_MEDIUM,
    onRequestClose: null,
    onScroll: null,
    containerProps: {},
    actionBarProps: {},
    contentProps: {},
    bottomActionBarProps: {}
  };

  state = {
    isOpened: true,
    actionBarShadow: false,
    bottomActionBarShadow: false
  };

  static getDerivedStateFromProps(props, state) {
    if (props.isOpened !== state.isOpened) {
      return {
        isOpened: props.isOpened
      };
    }

    return null;
  }

  componentDidMount() {
    if (this.props.isOpened) {
      this.updateModalHeight();
      this.updateShadows();
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.isOpened && !prevProps.isOpened) {
      this.updateModalHeight();
      this.updateShadows();
    }
  }

  actionBarRef = React.createRef();
  containerRef = React.createRef();
  contentRef = React.createRef();
  bottomActionBarRef = React.createRef();

  closeModal = event => {
    const { onRequestClose } = this.props;

    isRequestCloseApproved({ onRequestClose, event }).then(requestApproved => {
      if (requestApproved) {
        this.setState({ isOpened: false });
      }
    });
  };

  handleScroll = e => {
    const { onScroll } = this.props;

    if (onScroll) {
      onScroll(e);
    }

    this.updateShadows();
  };

  updateModalHeight = () => {
    const {
      deviceSize: { isSmall }
    } = this.props;

    const actionBar = this.actionBarRef.current;
    const bottomActionBar = this.bottomActionBarRef.current;
    const content = this.contentRef.current;
    const container = this.containerRef.current;

    const contentHeight = getContentHeight({
      actionBar,
      bottomActionBar,
      container
    });

    if (isSmall) {
      content.style.minHeight = contentHeight;
    }

    content.style.maxHeight = contentHeight;
  };

  updateShadows = () => {
    this.updateActionBarShadow();

    const bottomActionBar = this.bottomActionBarRef.current;

    if (bottomActionBar) {
      this.updateBottomActionBarShadow();
    }
  };

  updateActionBarShadow = () => {
    const content = this.contentRef.current;

    const actionBarShadow = getActionBarShadow({ content });

    if (actionBarShadow !== this.state.actionBarShadow) {
      this.setState({
        actionBarShadow
      });
    }
  };

  updateBottomActionBarShadow = () => {
    const content = this.contentRef.current;

    const bottomActionBarShadow = getBottomActionBarShadow({ content });

    if (bottomActionBarShadow !== this.state.bottomActionBarShadow) {
      this.setState({
        bottomActionBarShadow
      });
    }
  };

  render() {
    const {
      actionBar,
      bottomActionBar,
      children,
      gutters,
      size,
      deviceSize,
      containerProps,
      actionBarProps,
      contentProps,
      bottomActionBarProps,
      displayTop
    } = this.props;
    const { isOpened, actionBarShadow, bottomActionBarShadow } = this.state;
    const { closeModal } = this;

    return (
      <ModalProvider value={{ closeModal }}>
        <CSSTransition
          in={isOpened}
          key="overlay-animation"
          timeout={300}
          classNames="open"
        >
          <Backdrop
            childRef={this.containerRef}
            onRequestClose={this.closeModal}
            isVisible={isOpened}
            animated
          >
            <CSSTransition
              in={isOpened}
              key="modal-animation"
              timeout={300}
              classNames="open"
            >
              <ModalContainer
                small={getModalSize({ deviceSize, preferredSize: size })}
                ref={this.containerRef}
                isOpened={isOpened}
                size={size}
                displayTop={displayTop}
                {...containerProps}
              >
                {actionBar && (
                  <ActionBar
                    shadow={actionBarShadow}
                    ref={this.actionBarRef}
                    gutters={gutters}
                    {...actionBarProps}
                  >
                    {actionBar}
                  </ActionBar>
                )}
                <ModalContent
                  ref={this.contentRef}
                  onScroll={this.handleScroll}
                  gutters={gutters}
                  {...contentProps}
                >
                  {children}
                </ModalContent>
                {bottomActionBar && (
                  <BottomActionBar
                    shadow={bottomActionBarShadow}
                    ref={this.bottomActionBarRef}
                    gutters={gutters}
                    {...bottomActionBarProps}
                  >
                    {bottomActionBar}
                  </BottomActionBar>
                )}
              </ModalContainer>
            </CSSTransition>
          </Backdrop>
        </CSSTransition>
      </ModalProvider>
    );
  }
}

export default withDeviceSize(Modal);
