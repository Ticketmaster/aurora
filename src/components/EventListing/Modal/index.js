import React, { PureComponent, Fragment } from 'react';

import ModalTrigger from './ModalTrigger';
import ModalContent from './ModalContent';
import ModalBottomSheet from "./ModalBottomSheet";

class Modal extends PureComponent {
  state = { isOpen: false };

  // static getDerivedStateFromProps(nextProps, prevState) {
  //   console.log("getDerivedStateFromProps", nextProps);
  //   if (prevState.isOpen !== nextProps.isOpen) {
  //     return {
  //       isOpen: nextProps.isOpen
  //     };
  //   }

  //   return null;
  // }

  onOpen = (e) => {
    this.setState({ isOpen: true }, () => {
      this.closeButtonNode.focus(); // removes focus for screen readers
    });
    this.setState({ isOpen: true });
    this.toggleScrollLock();
    this.onOpenSuccess(e)
  };

  onClose = (e) => {
    this.setState({ isOpen: false });
    this.toggleScrollLock();
    this.handleOnCloseSuccess(e);
  };

  onClickAway = (e) => {
    if (this.modalNode && this.modalNode.contains(e.target)) return;
    this.onClose(e);
  };

  componentDidCatch() {
    this.setState(state => ({...state, hasError: true}))
  }

  onOpenSuccess = e => {
    const { onOpenSuccess } = this.props;
    if (onOpenSuccess) {
      onOpenSuccess(e);
    }
  }

  handleOnCloseSuccess = (e) => {
    const { onCloseSuccess } = this.props;
    if (onCloseSuccess){
      onCloseSuccess(e);
    }
  }

  toggleScrollLock = () => document.querySelector('html').classList.toggle('u-lock-scroll');
  
   clonedChildren = () => {
    const { state: { isOpen } } = this;

    const children = React.Children.map(this.props.children, (child) => {
        if (child && child.type.name === "ModalTrigger") {
            return React.cloneElement(
              child,
              {
                onOpen: (e) => this.onOpen(e),
                onClose: (e) => this.onClose(e),
                onOpenSuccess: this.onOpenSuccess,
                triggerRef: n => {this.openButtonNode = n}
              }
            );

        } else if (
          child.type.name === "ModalContent" ||
          child.type.name === "ModalBottomSheet"
          && isOpen
        ) {
           return ( 
            React.cloneElement(
              child,
              {
                  onOpen: (e) => this.onOpen(e),
                  buttonRef: node => {this.closeButtonNode = node},
                  modalRef: node => {this.modalNode = node},
                  onClickAway: (e) => this.onClickAway(e),
                  onClose: (e) => this.onClose(e)
              }
            )
           )
        }
        return children;
    });
    return children;
  };

  render() {
    const { isOpen, hasError } = this.state;
    const { ariaLabel, children, trigger, role } = this.props;

    if (hasError){
      return <p>Uh oh Something went wrong</p>
    } 
    return (
      <Fragment>
          {this.clonedChildren()}
      </Fragment>
    )
    
  }
}

Modal.Trigger = ModalTrigger;
Modal.BottomSheet = ModalBottomSheet;
Modal.Content = ModalContent;

export default Modal;