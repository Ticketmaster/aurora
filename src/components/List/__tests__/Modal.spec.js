import React from "react";
import { render, Simulate } from "react-testing-library";

import Modal from "../Modal";

describe("<Modal />", () => {
  beforeEach(() => {
    const modalRoot = global.document.createElement("div");
    modalRoot.setAttribute("id", "modal-root");
    modalRoot.setAttribute("data-testid", "modal-root");
    const body = global.document.querySelector("body");
    body.appendChild(modalRoot);
  });

  afterEach(() => {
    const body = global.document.querySelector("body");
    const modalRoot = global.document.getElementById("modal-root");
    body.removeChild(modalRoot);
  });

  it("renders Modal correctly", () => {
    const { container } = render(
      <Modal>
        <div>Europe</div>
        <div>Africa</div>
        <div>Asias</div>
      </Modal>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("removes the Modal on clicking cancel button", () => {
    const { container } = render(
      <Modal>
        <div>Europe</div>
        <div>Africa</div>
        <div>Asias</div>
      </Modal>
    );
    Simulate.click(container.querySelector(".button--close"));
    expect(container.innerHTML).toMatchSnapshot();
  });

  it("should not display the cancel button", () => {
    const { container } = render(
      <Modal withCloseIcon={false}>
        <div>Europe</div>
        <div>Africa</div>
        <div>Asias</div>
      </Modal>
    );
    expect(container.querySelector(".button--close")).toEqual(null);
    expect(container.innerHTML).toMatchSnapshot();
  });
});
