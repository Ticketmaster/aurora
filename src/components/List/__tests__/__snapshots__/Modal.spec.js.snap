// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> removes the Modal on clicking cancel button 1`] = `"<div role=\\"dialog\\" aria-modal=\\"true\\" class=\\"sc-ifAKCX dYcmid\\"><div class=\\"sc-bZQynM gRHrxt\\"><button size=\\"45\\" class=\\"button--close sc-htpNat ckFINe\\" aria-label=\\"Close Modal\\" role=\\"button\\"><svg viewBox=\\"0 0 12 12\\" width=\\"12\\" height=\\"12\\" fill=\\"rgba(38, 38, 38, 1)\\"><path d=\\"M6.563 6.203l4.523 4.516a.384.384 0 0 1 .117.281c0 .11-.039.203-.117.281a.378.378 0 0 1-.137.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.136-.09L6 6.766 1.484 11.28a.378.378 0 0 1-.136.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.137-.09A.384.384 0 0 1 .797 11c0-.11.039-.203.117-.281l4.524-4.516L.913 1.68a.384.384 0 0 1-.117-.282c0-.109.039-.203.117-.28A.388.388 0 0 1 1.2 1c.112 0 .207.04.285.117L6 5.633l4.516-4.516A.388.388 0 0 1 10.8 1c.112 0 .207.04.285.117a.384.384 0 0 1 .117.281c0 .11-.039.204-.117.282L6.562 6.203z\\"></path></svg></button></div><div class=\\"sc-EHOje QamKp\\"><div>Europe</div><div>Africa</div><div>Asias</div></div></div>"`;

exports[`<Modal /> renders Modal correctly 1`] = `
.c2 {
  height: 36px;
  width: 100%;
  min-width: 100px;
  font-weight: 600;
  text-align: center;
  border-radius: 2px;
  color: rgba(255,255,255,1);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,1);
  cursor: pointer;
  font-size: 14px;
  min-width: 45px;
  max-width: 45px;
  height: 45px;
  width: 45px;
  border: none;
}

.c2:disabled {
  color: rgba(255,255,255,0.5);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,0.5);
  cursor: not-allowed;
}

.c2:focus {
  outline: none;
}

.c2 > * {
  pointer-events: none;
}

.c0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 400px;
  max-width: 640px;
  position: relative;
  background-color: rgba(255,255,255,1);
  border-radius: 4px;
  box-shadow: 0 16px 16px 0 rgba(0,0,0,0.06),0 0 16px 0 rgba(0,0,0,0.12);
  border: solid 1px rgba(0,0,0,0.04);
  padding: 0;
  overflow-y: auto;
  overflow-x: hidden;
  max-height: calc(100% - 96px);
  z-index: 100;
}

.modal-enter .c0 {
  opacity: 0;
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.modal-leave .c0,
.modal-exit .c0 {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.modal-appear .c0 {
  opacity: 0;
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.modal-enter.modal-enter-active .c0 {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.modal-leave.modal-leave-active .c0,
.modal-exit.modal-exit-active .c0 {
  opacity: 0;
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.modal-appear.modal-appear-active .c0 {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c3 {
  padding: 0 32px 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  margin-left: -8px;
  margin-right: -8px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding-right: 16px;
}

.c1 .row--gutterless {
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
  margin-left: 0;
  margin-right: 0;
}

@media screen and (min-width:768px) {
  .c1 {
    margin-left: -12px;
    margin-right: -12px;
  }
}

<div
  aria-modal="true"
  class="c0"
  role="dialog"
>
  <div
    class="c1"
  >
    <button
      aria-label="Close Modal"
      class="button--close c2"
      role="button"
      size="45"
    >
      <svg
        fill="rgba(38, 38, 38, 1)"
        height="12"
        viewBox="0 0 12 12"
        width="12"
      >
        <path
          d="M6.563 6.203l4.523 4.516a.384.384 0 0 1 .117.281c0 .11-.039.203-.117.281a.378.378 0 0 1-.137.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.136-.09L6 6.766 1.484 11.28a.378.378 0 0 1-.136.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.137-.09A.384.384 0 0 1 .797 11c0-.11.039-.203.117-.281l4.524-4.516L.913 1.68a.384.384 0 0 1-.117-.282c0-.109.039-.203.117-.28A.388.388 0 0 1 1.2 1c.112 0 .207.04.285.117L6 5.633l4.516-4.516A.388.388 0 0 1 10.8 1c.112 0 .207.04.285.117a.384.384 0 0 1 .117.281c0 .11-.039.204-.117.282L6.562 6.203z"
        />
      </svg>
    </button>
  </div>
  <div
    class="c3"
  >
    <div>
      Europe
    </div>
    <div>
      Africa
    </div>
    <div>
      Asias
    </div>
  </div>
</div>
`;

exports[`<Modal /> should not display the cancel button 1`] = `"<div role=\\"dialog\\" aria-modal=\\"true\\" class=\\"sc-ifAKCX dYcmid\\"><div class=\\"sc-EHOje QamKp\\"><div>Europe</div><div>Africa</div><div>Asias</div></div></div>"`;
